src/iface/ethernet.rs

```diff
@@ -61,72 +66,54 @@ impl<'a, 'b, 'c, DeviceT: for<'d> Device<'d>> Interface<'a, 'b, 'c, DeviceT> {
                   ProtocolAddrsMT: Into<ManagedSlice<'c, IpCidr>>,
                   Ipv4GatewayAddrT: Into<Option<Ipv4Address>>, {
         let device = device.into();
-        let arp_cache = arp_cache.into();
-        let ip_addrs = ip_addrs.into();
-        let ipv4_gateway = ipv4_gateway.into();
-
-        Self::check_ethernet_addr(&ethernet_addr);
-        Self::check_ip_addrs(&ip_addrs);
-        Interface { device, arp_cache, ethernet_addr, ip_addrs, ipv4_gateway}
-    }
+        let caps = device.capabilities();
+        let inner = InterfaceInner::new(arp_cache, ethernet_addr, ip_addrs, ipv4_gateway, caps);
```

Comment:
This is an implementation detail so just initialize it with struct initialization syntax. No need for a separate `new` method etc.